src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.cs

```diff
@@ -656,8 +654,16 @@ public ResultWrapper<IEnumerable<FilterLog>> eth_getLogs(Filter filter)
                 return ResultWrapper<IEnumerable<FilterLog>>.Fail($"'From' block '{fromBlockNumber}' is later than 'to' block '{toBlockNumber}'.", ErrorCodes.InvalidParams);
             }
 
-            return ResultWrapper<IEnumerable<FilterLog>>.Success(GetLogs(filter.FromBlock, filter.ToBlock,
-                cancellationTokenSource, cancellationToken));
+            try
+            {
+                IEnumerable<FilterLog> filterLogs = _blockchainBridge.GetLogs(filter.FromBlock, filter.ToBlock,
+                    filter.Address, filter.Topics, cancellationToken);
+                return ResultWrapper<IEnumerable<FilterLog>>.Success(GetLogs(filterLogs, cancellationTokenSource));
+            }
+            catch (ResourceNotFoundException exception)
+            {
+                return ResultWrapper<IEnumerable<FilterLog>>.Fail(exception.Message, ErrorCodes.ResourceNotFound);
+            }
```

Comment:
Generally this is lazy collection, does it throw without enumeration (maybe yes as its before first yield)?

Reply:
It should throw before the iterator is returned.

Reply:
As in, the line 659, not in line 618.

Reply:
But its still lazy right?

Reply:
The iteration is still lazy. I move the call outside the function specifically so that it throws outside iteration. Otherwise, the exception will not be caught.